safe-global/safe-wallet-web
# fix: only delete all preferences per chain

## What it solves

Resolves https://github.com/safe-global/safe-wallet-web/pull/2546#issuecomment-1736499323

## How this PR fixes it

`deleteAllPreferences` is has been replaced with `deleteAllChainPreferences`, which clears preferences according to a provided `chainId` according to the device unregistration (also per `chainId`).

A loading indicator has also been added to the global preferences "Save" button.

## How to test it

During the following registrations, a loading indicator should appear in the "Save" button:

1. Ensure multiple Safes are registered across differing chains.
2. Navigate to the global notification preferences.
3. Unregister a Safe.
4. Observe _only_ the preferences from that Safe removed.
---

1. Ensure multiple Safes are registered across differing chains.
2. Navigate to the global notification preferences.
3. Unregister _all_ Safes on one network.
6. Observe the Safes on other chain(s) retain their preferences.

## Checklist
